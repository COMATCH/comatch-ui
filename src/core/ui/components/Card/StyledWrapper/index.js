export * from './StyledHeader';
export * from './StyledWrapper';
