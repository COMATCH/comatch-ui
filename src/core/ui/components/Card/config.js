const BACKGROUND_COLOR = {
    BLACK: 'black',
    WHITE: 'white',
    LIGHT_BLUE: 'lightBlue',
    PRIMARY: 'primary',
    LIGHT_PRIMARY: 'lightPrimary',
    SECONDARY: 'secondary',
    LIGHT_GRAY: 'lightGray',
    MID_GRAY: 'midGray',
    GRAY: 'gray',
    DARK_GRAY: 'darkGray',
    GREEN: 'green',
    LIGHT_GREEN: 'lightGreen',
    DARK_GREEN: 'darkGreen',
    RED: 'red',
    LIGHT_RED: 'lightRed',
    DARK_RED: 'darkRed',
    LIGHT_ORANGE: 'lightOrange',
    ORANGE: 'orange',
    DARK_ORANGE: 'darkOrange',
    YELLOW: 'yellow',
    TRANSPARENT: 'transparent',
};

const FONT_COLOR = {
    BLACK: 'white',
    WHITE: 'black',
    LIGHT_BLUE: 'black',
    PRIMARY: 'white',
    LIGHT_PRIMARY: 'black',
    SECONDARY: 'white',
    LIGHT_GRAY: 'black',
    MID_GRAY: 'black',
    GRAY: 'white',
    DARK_GRAY: 'white',
    GREEN: 'white',
    LIGHT_GREEN: 'black',
    DARK_GREEN: 'white',
    RED: 'white',
    LIGHT_RED: 'black',
    DARK_RED: 'white',
    LIGHT_ORANGE: 'black',
    ORANGE: 'black',
    DARK_ORANGE: 'white',
    YELLOW: 'black',
    TRANSPARENT: 'black',
};

const ORIENTATION = {
    HORIZONTAL: 'horizontal',
    VERTICAL: 'vertical',
};

export { BACKGROUND_COLOR, FONT_COLOR, ORIENTATION };
