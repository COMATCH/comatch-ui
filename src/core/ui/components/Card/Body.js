import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { StyledBody } from './StyledWrapper';
import { BACKGROUND_COLOR, ORIENTATION } from './config';

const propTypes = {
    backgroundColor: PropTypes.oneOf(Object.keys(BACKGROUND_COLOR)),
    children: PropTypes.node,
    className: PropTypes.string,
    id: PropTypes.string,
    orientation: PropTypes.oneOf(Object.values(ORIENTATION)),
    wrap: PropTypes.bool,
};

const defaultProps = {
    backgroundColor: 'TRANSPARENT',
    children: null,
    className: null,
    id: null,
    orientation: ORIENTATION.VERTICAL,
    wrap: false,
};

const Body = ({ backgroundColor, children, className, id, orientation, wrap }) =>
    !!children && (
        <StyledBody
            backgroundColor={backgroundColor}
            className={classnames('Body', className)}
            {...(id && { id })}
            orientation={orientation}
            wrap={wrap}
        >
            {children}
        </StyledBody>
    );

Body.propTypes = propTypes;
Body.defaultProps = defaultProps;

export { Body };
