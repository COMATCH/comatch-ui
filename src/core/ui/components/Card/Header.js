import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { StyledHeader, StyledHeaderControls } from './StyledWrapper';
import { BACKGROUND_COLOR } from './config';
import { Splash } from '../Splash';

const propTypes = {
    backgroundColor: PropTypes.oneOf(Object.keys(BACKGROUND_COLOR)),
    children: PropTypes.node,
    className: PropTypes.string,
    controls: PropTypes.node,
    id: PropTypes.string,
    title: Splash.propTypes.title,
    titleOrientation: Splash.propTypes.orientation,
    subtitle: Splash.propTypes.subtitle,
    wrap: PropTypes.bool,
};

const defaultProps = {
    backgroundColor: 'TRANSPARENT',
    children: null,
    className: null,
    controls: null,
    id: null,
    title: Splash.defaultProps.title,
    titleOrientation: Splash.defaultProps.orientation,
    subtitle: Splash.defaultProps.subtitle,
    wrap: false,
};

const Header = ({ backgroundColor, children, className, controls, id, title, titleOrientation, subtitle, wrap }) =>
    (!!children || !!title || !!subtitle || !!controls) && (
        <StyledHeader
            backgroundColor={backgroundColor}
            className={classnames('Header', className)}
            {...(id && { id })}
            wrap={wrap}
        >
            <Splash orientation={titleOrientation} title={title} subtitle={subtitle} />
            {children}
            <StyledHeaderControls>{controls}</StyledHeaderControls>
        </StyledHeader>
    );

Header.propTypes = propTypes;
Header.defaultProps = defaultProps;

export { Header };
