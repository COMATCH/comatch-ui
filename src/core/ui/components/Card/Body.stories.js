import React from 'react';
import styled from 'styled-components';
import Highlight from 'react-highlight';
import { storiesOf } from '@storybook/react';
import { Card } from './Card';
import { BACKGROUND_COLOR, ORIENTATION } from './config';

const Collection = styled.div`
    display: flex;
    flex-flow: row wrap;
    justify-content: space-evenly;
    min-height: 125px;

    .Body {
        flex: 450px 0;
        margin: 20px;
    }
`;

const BasicContent = styled.div`
    padding: 10px 15px;
`;

const Content = (
    <>
        <div>Item 1</div>
        <div>Item 2</div>
    </>
);

storiesOf('Card|Card.Body', module)
    .add('default', () => (
        <>
            <Card.Body />
            <Highlight className="html">{'<Card.Body />'}</Highlight>
        </>
    ))
    .add('with different `orientation`', () => (
        <>
            <Collection>
                <Card.Body backgroundColor="primary" orientation={ORIENTATION.HORIZONTAL}>
                    {Content}
                </Card.Body>
                <Card.Body backgroundColor="primary" orientation={ORIENTATION.VERTICAL}>
                    {Content}
                </Card.Body>
            </Collection>
            <Highlight className="html">
                {'<Card.Body backgroundColor="PRIMARY" orientation={ORIENTATION.VERTICAL}>{Content}</Card.Body>' +
                    '\n<Card.Body backgroundColor="PRIMARY" orientation={ORIENTATION.VERTICAL}>{Content}</Card.Body>'}
            </Highlight>
        </>
    ))
    .add('with different `backgroundColor`', () => (
        <>
            <Collection>
                {Object.values(BACKGROUND_COLOR).map((backgroundColor) => (
                    <Card.Body key={backgroundColor} backgroundColor={backgroundColor}>
                        <BasicContent>{backgroundColor}</BasicContent>
                    </Card.Body>
                ))}
            </Collection>
            <Highlight className="html">
                {Object.keys(BACKGROUND_COLOR).map(
                    (backgroundColor, index) =>
                        `${index ? '\n\n' : ''}<Card.Body backgroundColor={BACKGROUND_COLOR.${backgroundColor}}>` +
                        `\n\t<BasicContent>{${backgroundColor}}</BasicContent>` +
                        `\n</Card.Body>`,
                )}
            </Highlight>
        </>
    ));
