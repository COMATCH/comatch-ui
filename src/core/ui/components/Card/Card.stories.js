import React from 'react';
import styled from 'styled-components';
import Highlight from 'react-highlight';
import { storiesOf } from '@storybook/react';
import { Card } from './Card';
import { Button } from '../Button';
import { BACKGROUND_COLOR } from './config';

const Collection = styled.div`
    display: flex;
    flex-flow: row wrap;
    justify-content: space-evenly;
    min-height: 125px;

    .Card {
        flex: 500px 0;
        margin: 20px;
    }
`;

const BasicContent = styled.div`
    padding: 10px 15px;
`;

const actions = (
    <>
        <Button text="Action A" ghost />
        <Button text="Action B" />
    </>
);

storiesOf('Card|Card.Card', module)
    .add('default', () => (
        <>
            <Card />
            <Highlight className="html">{'<Card />'}</Highlight>
        </>
    ))
    .add('with different `backgroundColor`', () => (
        <>
            <Collection>
                {Object.keys(BACKGROUND_COLOR).map((backgroundColor) => (
                    <Card key={backgroundColor} backgroundColor={backgroundColor}>
                        <BasicContent>{backgroundColor}</BasicContent>
                    </Card>
                ))}
            </Collection>
            <Highlight className="html">
                {Object.keys(BACKGROUND_COLOR).map(
                    (backgroundColor, index) =>
                        `${index ? '\n\n' : ''}<Card backgroundColor={${backgroundColor}}>` +
                        `\n\t<BasicContent>{${backgroundColor}}</BasicContent>` +
                        `\n</Card>`,
                )}
            </Highlight>
        </>
    ))
    .add('with `Header` & `Body`', () => (
        <>
            <Collection>
                {Object.keys(BACKGROUND_COLOR).map((backgroundColor) => (
                    <Card key={backgroundColor} backgroundColor={backgroundColor}>
                        <Card.Header title="Header" subtitle="Header subtitle" controls={actions} />
                        <Card.Body>
                            <p>This is the Card&apos;s Body (Content)</p>
                        </Card.Body>
                    </Card>
                ))}
            </Collection>
            <Highlight className="html">
                {Object.keys(BACKGROUND_COLOR).map(
                    (backgroundColor, index) =>
                        `${index ? '\n\n' : ''}<Card backgroundColor={${backgroundColor}}>` +
                        '\n\t<Card.Header title="Header" subtitle="Header subtitle" controls={actions} />' +
                        "\n\t<Card.Body>\n\t\t<p>This is the Card's Body (Content)</p>" +
                        '\n\t</Card.Body>' +
                        '\n</Card>',
                )}
            </Highlight>
        </>
    ));
