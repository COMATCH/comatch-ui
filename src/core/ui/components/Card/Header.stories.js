import React from 'react';
import styled from 'styled-components';
import Highlight from 'react-highlight';
import { storiesOf } from '@storybook/react';
import { Card } from './Card';
import { BACKGROUND_COLOR } from './config';
import { ORIENTATION } from '../Splash';

const Collection = styled.div`
    display: flex;
    flex-flow: row wrap;
    justify-content: space-evenly;
    min-height: 125px;

    .Header {
        flex: 450px 0;
        margin: 20px;
    }
`;

const BasicContent = styled.div`
    padding: 10px 15px;
`;

const BasicControls = (
    <>
        <button>Button 1</button>
        <button>Button 2</button>
        <button>Button 3</button>
    </>
);

storiesOf('Card|Card.Header', module)
    .add('default', () => (
        <>
            <Card.Header />
            <Highlight className="html">{'<Card.Header />'}</Highlight>
        </>
    ))
    .add('with different `backgroundColor`', () => (
        <>
            <Collection>
                {Object.keys(BACKGROUND_COLOR).map((backgroundColor) => (
                    <Card.Header key={backgroundColor} backgroundColor={backgroundColor}>
                        <BasicContent>{backgroundColor}</BasicContent>
                    </Card.Header>
                ))}
            </Collection>
            <Highlight className="html">
                {Object.keys(BACKGROUND_COLOR).map(
                    (backgroundColor, index) =>
                        `${index ? '\n\n' : ''}<Card.Header backgroundColor={${backgroundColor}}>` +
                        `\n\t<BasicContent>{${backgroundColor}}</BasicContent>` +
                        `\n</Card.Header>`,
                )}
            </Highlight>
        </>
    ))
    .add('with `controls`', () => (
        <>
            <Card.Header controls={BasicControls} />
            <Highlight className="html">{'<Card.Header controls={BasicControls} />'}</Highlight>
        </>
    ))
    .add('with content, `controls` and different `backgroundColor`', () => (
        <>
            <Collection>
                {Object.keys(BACKGROUND_COLOR).map((backgroundColor) => (
                    <Card.Header key={backgroundColor} backgroundColor={backgroundColor} controls={BasicControls}>
                        <BasicContent>{backgroundColor}</BasicContent>
                    </Card.Header>
                ))}
            </Collection>
            <Highlight className="html">
                {Object.keys(BACKGROUND_COLOR).map(
                    (backgroundColor, index) =>
                        `${
                            index ? '\n\n' : ''
                        }<Card.Header backgroundColor={${backgroundColor}} controls={BasicControls}>` +
                        `\n\t<BasicContent>{${backgroundColor}}</BasicContent>` +
                        `\n</Card.Header>`,
                )}
            </Highlight>
        </>
    ))
    .add('with `title` & `subtitle`', () => (
        <>
            <Card.Header title="Title" titleOrientation={ORIENTATION.VERTICAL} subtitle="Subtitle" />
            <Highlight className="html">
                {'<Card.Header title="Title" titleOrientation={ORIENTATION.VERTICAL} subtitle="Subtitle" />'}
            </Highlight>
        </>
    ))
    .add('with content, `title`, `subtitle` and different `backgroundColor`', () => (
        <>
            <Collection>
                {Object.keys(BACKGROUND_COLOR).map((backgroundColor) => (
                    <Card.Header
                        key={backgroundColor}
                        backgroundColor={backgroundColor}
                        title="Title"
                        titleOrientation={ORIENTATION.VERTICAL}
                        subtitle="Subtitle"
                    >
                        <BasicContent>{backgroundColor}</BasicContent>
                    </Card.Header>
                ))}
            </Collection>
            <Highlight className="html">
                {Object.keys(BACKGROUND_COLOR).map(
                    (backgroundColor, index) =>
                        `${index ? '\n\n' : ''}<Card.Header backgroundColor={${backgroundColor}} ` +
                        `title="Title" titleOrientation={ORIENTATION.VERTICAL} subtitle="Subtitle">` +
                        `\n\t<BasicContent>{${backgroundColor}}</BasicContent>` +
                        `\n</Card.Header>`,
                )}
            </Highlight>
        </>
    ));
