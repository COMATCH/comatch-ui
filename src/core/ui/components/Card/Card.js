import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { Header } from './Header';
import { BACKGROUND_COLOR } from './config';
import { StyledWrapper } from './StyledWrapper';

const propTypes = {
    backgroundColor: PropTypes.oneOf(Object.keys(BACKGROUND_COLOR)),
    bordered: PropTypes.bool,
    className: PropTypes.string,
    children: PropTypes.node,
    id: PropTypes.string,
    inline: PropTypes.bool,
};

const defaultProps = {
    backgroundColor: 'WHITE',
    bordered: false,
    className: null,
    children: null,
    id: null,
    inline: false,
};

const Card = ({ backgroundColor, bordered, children, className, id, inline }) =>
    !!children && (
        <StyledWrapper
            backgroundColor={backgroundColor}
            bordered={bordered}
            className={classnames('Card', className)}
            inline={inline}
            {...(id && { id })}
        >
            {children}
        </StyledWrapper>
    );

Card.propTypes = propTypes;
Card.defaultProps = defaultProps;

Card.Header = Header;

export { Card };
