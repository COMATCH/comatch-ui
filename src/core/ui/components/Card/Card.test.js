import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { Card } from './Card';
import { Button } from '../Button';
import { BACKGROUND_COLOR } from './config';
import '../../../config/tests/setup';

describe('Card', () => {
    it('should render the default Card correctly', () => {
        const card = renderer.create(<Card>card</Card>).toJSON();
        expect(card).toMatchSnapshot();
    });

    it('should render a customized Card correctly', () => {
        const card = renderer
            .create(
                <Card backgroundColor={BACKGROUND_COLOR.PRIMARY} bordered className="ACard" id="a-card" inline>
                    <Card.Header title="Header" subtitle="Header subtitle" controls={<Button>BTN</Button>} />
                    <Card.Body>
                        <p>This is the Card&apos;s Body (Content)</p>
                    </Card.Body>
                    <div>some content...</div>
                </Card>,
            )
            .toJSON();
        expect(card).toMatchSnapshot();
    });

    it('should have the correct default props', () => {
        const card = mount(<Card />);
        expect(card.prop('backgroundColor')).toEqual('white');
        expect(card.prop('bordered')).toEqual(false);
        expect(card.prop('className')).toEqual(null);
        expect(card.prop('id')).toEqual(null);
        expect(card.prop('inline')).toEqual(false);
    });
});
