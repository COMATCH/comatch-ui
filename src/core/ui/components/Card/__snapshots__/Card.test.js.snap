// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render a customized Card correctly 1`] = `
<div
  className="Card ACard sc-ifAKCX goZsUg"
  id="a-card"
>
  <div
    className="Header sc-bxivhb iczkCS"
  >
    <div
      className="Splash TitleAndSubtitle sc-gzVnrw bpmeZP"
    >
      <div
        className="Title sc-bZQynM bPbNzY"
      >
        Header
      </div>
      <div
        className="Subtitle sc-EHOje jUaBYT"
      >
        Header subtitle
      </div>
    </div>
    <div
      className="Controls sc-htpNat cPrsnq"
    >
      <button
        className="Button rectangle full only-icon sc-iwsKbI dlgvgE"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        <span>
          
        </span>
      </button>
    </div>
  </div>
  <div
    className="Body sc-bdVaJa jJZBvK"
  >
    <p>
      This is the Card's Body (Content)
    </p>
  </div>
  <div>
    some content...
  </div>
</div>
`;

exports[`Card should render the default Card correctly 1`] = `
<div
  className="Card sc-ifAKCX jzAoDZ"
>
  card
</div>
`;
